### PR TITLE
Feat: Explicitly add Orqusetra-sdk dependency to IR

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
Previously we were adding orquestra-sdk as dependnecy on server side. But we are removing the SDK installation on server side

# This PR's solution
move dependnecy-injection into client-side

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: projects/orquestra-sdk/docs/examples/tests/test_migrating_from_quantum_engine.py
Comment (jamesclark-Zapata) [inline projects/orquestra-sdk/docs/examples/tests/test_migrating_from_quantum_engine.py:1]: I'm assuming this is no longer used in the docs?

File: projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_traversal.py [hunk 1]
```diff
@@ -812,18 +824,24 @@ def flatten_graph(
         ]:
             if imp not in import_models_dict:
                 if isinstance(imp, _dsl.DeferredGitImport):
-                    cashe_key = (imp.local_repo_path, imp.git_ref)
-                    if cashe_key in cached_git_import_dict:
-                        import_models_dict[imp] = cached_git_import_dict[cashe_key]
+                    cache_key = (imp.local_repo_path, imp.git_ref)
```
Comment (jamesclark-Zapata) [inline projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_traversal.py:827]: 😅

**Review by jamesclark-Zapata — APPROVED** [review]

File: projects/orquestra-sdk/docs/examples/tests/test_migrating_from_quantum_engine.py
Comment (SebastianMorawiec) [reply projects/orquestra-sdk/docs/examples/tests/test_migrating_from_quantum_engine.py:1]: Yes, Ive removed the docs site a while ago, forgot about the tests

File: projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_traversal.py (hunk 1 above)
Comment (SebastianMorawiec) [reply projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_traversal.py:827]: 🤷 what can I say 😅

**Pull request merged**